File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr, int divisor) {
    vector<int> r;
    for(int i=0;i<arr.size();i++) {
        if(arr[i]%divisor==0)
            r.push_back(arr[i]);
    }
    sort(r.begin(),r.end(),less<int>());
    if(r.size()==0)
        r.push_back(-1);
    return r;
}

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] arr, int divisor) {
        int cnt=0;
        for(int i:arr) {
            if(i%divisor==0)
                cnt++;
        }
        if(cnt==0)
            return new int[]{-1};
        else {
            int[] rlt=new int[cnt];
            cnt=0;
            for(int j:arr) {
                if(j%divisor==0) {
                    rlt[cnt++]=j;
                }
            }
            Arrays.sort(rlt);
            return rlt;
        }
    }
}